src/sentry/api/endpoints/project_rule_preview.py

```diff
@@ -39,13 +42,18 @@ def post(self, request: Request, project) -> Response:
             raise ValidationError
 
         data = serializer.validated_data
+        try:
+            endpoint = parse_datetime(request.data["endpoint"])
```

Comment:
Should this be validated like other serializer fields are?

Reply:
This is a field that's passed internally between the frontend and the backend, and I'd probably need a lone serializer just for this field since it's specific to preview and not any other rule endpoint.

Reply:
added a serializer for this